### PR TITLE
Added support for the service monitor targetLabels

### PR DESCRIPTION
**Pull request by andrey-dubnik**
Hi, hope opening PR is Ok? This should add support for the ServiceMonitor target labels as per #1804

### CODE REVIEW

File: pkg/reconcile/pipeline/infinispan/handler/provision/service_monitor.go [hunk 1]
```diff
@@ -68,6 +68,7 @@ func ServiceMonitor(i *iv1.Infinispan, ctx pipeline.Context) {
 			NamespaceSelector: monitoringv1.NamespaceSelector{
 				MatchNames: []string{i.Namespace},
 			},
+			TargetLabels: i.ServiceMonitorTargetLabels(),
```
Comment (ryanemerson) [inline pkg/reconcile/pipeline/infinispan/handler/provision/service_monitor.go:71]: You should be able to remove this as the `mutateFn` will always be applied, even on initial create.

File: pkg/reconcile/pipeline/infinispan/pipeline/pipeline.go [hunk 2]
```diff
@@ -222,6 +222,7 @@ func (b *builder) Build() pipeline.Pipeline {
 	handlers.Add(
 		manage.ConsoleUrl,
 	)
+	handlers.AddFeatureSpecific(i.IsServiceMonitorEnabled(), provision.ServiceMonitor)
```
Comment (ryanemerson) [inline pkg/reconcile/pipeline/infinispan/pipeline/pipeline.go:225]: This should be removed as we already add this function to the pipeline  on L193.

**Review by ryanemerson — CHANGES_REQUESTED** [review]
> Hi, hope opening PR is Ok?

Absolutely, thank you for the contribution :slightly_smiling_face: 

There's just a couple of minor things and then we're good to go. I'll add some documentation in a follow up PR.

File: pkg/reconcile/pipeline/infinispan/handler/provision/service_monitor.go (hunk 1 above)
Comment (andrey-dubnik) [reply pkg/reconcile/pipeline/infinispan/handler/provision/service_monitor.go]: removed the line

File: pkg/reconcile/pipeline/infinispan/pipeline/pipeline.go (hunk 2 above)
Comment (andrey-dubnik) [reply pkg/reconcile/pipeline/infinispan/pipeline/pipeline.go]: removed the line

**Comment by andrey-dubnik** [comment]
@ryanemerson updated the code as per the suggestion

**Comment by ryanemerson** [comment]
Thanks @andrey-dubnik. I'll release 2.3.2.CSV with this fix, it should be available tomorrow.

**Comment by ryanemerson** [comment]
@andrey-dubnik The 2.3.2 Operator has been released and merged into the Operator Hub repo, it should be available to consume.

**Comment by andrey-dubnik** [comment]
Thanks a lot @ryanemerson !

**Pull request merged**